### PR TITLE
adding modes/levels to epubcheck (warning, error, fatal)

### PR DESCRIPTION
**Pull request by mekarpeles**

### CODE REVIEW

File: abbyy_to_epub3/create_epub.py [hunk 1]
```diff
@@ -128,7 +130,10 @@ def __init__(
 
         # Initialize all the book's variables cleanly
         self.debug = debug
-        self.args = args
+        self.epubcheck = epubcheck or (
+            # If no epubcheck specified and we're in debug mode, run
+            # --epubcheck warning
+            self.DEFAULT_EPUBCHECK_LEVEL if self.debug and not epubcheck else None)
```
Comment (bfalling) [inline abbyy_to_epub3/create_epub.py]: SUGGESTION: With short-circuit Boolean operators, I think `epubcheck` is already guaranteed to be false here. Remove from condition?

File: abbyy_to_epub3/create_epub.py [hunk 2]
```diff
@@ -1133,21 +1138,25 @@ def craft_epub(self, epub_outfile="out.epub", tmpdir=None):
         epub.write_epub(epub_outfile, self.book, {})
 
         # run validation on epub
-        if self.args and self.args.epubcheck:
-            self.validate_epub(epub_outfile)
+        if self.debug or (self.epubcheck):
```
Comment (bfalling) [inline abbyy_to_epub3/create_epub.py]: QUESTION: Parentheses needed around `self.epubcheck`?

File: abbyy_to_epub3/create_epub.py [hunk 3]
```diff
@@ -1133,21 +1138,25 @@ def craft_epub(self, epub_outfile="out.epub", tmpdir=None):
...
+            self.validate_epub(epub_outfile, level=self.epubcheck)
 
-    def validate_epub(self, epub_file):
+    def validate_epub(self, epub_file, level=None):
         self.logger.debug("Running EpubCheck on {}".format(epub_file))
+        LEVELS = ['warning', 'error', 'fatal']
+        level = level.lower() or self.DEFAULT_EPUBCHECK_LEVEL
+        try:
+            desired_levels = LEVELS[LEVELS.index(level):]
```
Comment (bfalling) [inline abbyy_to_epub3/create_epub.py]: QUESTION: This seems a little "clever". Why not just check to see if `LEVELS` contains the `level`?

**Comment by mekarpeles** [comment]
re: https://github.com/internetarchive/epub3/pull/13#discussion_r183555212
The levels are sorted and accumulative. Meaning "warning" would result in a check for all 3 levels whereas "error" would only check for "error" and "fatal", etc

**Pull request merged**